app/services/versioned_files_service/contents.rb

```diff
@@ -0,0 +1,26 @@
+class VersionedFilesService
+  # Support for managing content files.
+  class Contents
+    def initialize(service:)
+      @service = service
+    end
+
+    # @return [Array<String>] the md5s for all content files
+    def content_md5s
+      return [] unless content_path.exist?
+
+      content_path.children.map { |child| child.basename.to_s }
+    end
+
+    def move_content(md5:, source_path:)
+      FileUtils.mkdir_p(content_path)
+      FileUtils.mv(source_path, content_path_for(md5:))
+    end
+
+    def delete_content(md5:)
```

Comment:
Can this move to `PurgeContentAction`? It's only used there and we have to delegate to it a bunch of times.